Code/Framework/AzFramework/AzFramework/Asset/AssetCatalog.cpp

```diff
@@ -810,6 +810,16 @@ namespace AzFramework
     // the results.  For this reason, it is a direct call and protects its structures with a lock.
     void AssetCatalog::AssetChanged(const AZStd::vector<AzFramework::AssetSystem::AssetNotificationMessage>& messages, bool isCatalogInitialize)
     {
+        if (isCatalogInitialize && !m_initialized)
+        {
+            // This can happen when running with VFS, where the AP connection is done first
+            // and it's too early to send asset changed messages since catalog is not initialized yet.
+            AZ_TracePrintf(
+                "AssetCatalog",
+                "Skipping AssetChanged to update the catalog since it's too early, catalog has not been initialized yet.");
+            return;
+        }
```

Comment:
@AMZN-stankowi @amzn-mike Will this cause assets not in the AssetCatalog.xml on the host machine within the `<project-path>/Cache/android` directory to not send over the asset changed events if the AP happened to have process new android assets during it's run (because it had `android` platform enabled), before the remote Android system connected to it?

Reply:
Not sure that's an scenario expected to work for android at this time. Depending on the packing mode used:
- Assets in APK: the android assets must be fully generated before building android.
- Assets in sdcard: the android assets must be fully generated before deploying to device.
- Assets by VFS: the minimum assets (.setreg files) must be generated in the cache before deploying to device (otherwise it'll crash), and any the other assets are picked up at runtime when the app is launched, read by the host. I'd say that protecting against the host not processed the android asset when android app is launched is a bit of a stretched scenario to worry about at this point.